src/actions/jsmd-parser.js

```diff
@@ -1,20 +1,49 @@
+function hashCode(str) {
+  // this is an implementation of java's hashcode method
+  // https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript
+  let hash = 0
+  let chr
+  if (str.length === 0) return hash;
+  for (let i = 0; i < str.length; i++) {
+    chr = str.charCodeAt(i);
+    hash = ((hash << 5) - hash) + chr // eslint-disable-line
+    hash |= 0 // eslint-disable-line
+  }
+  return hash.toString()
+}
+
 export function jsmdParser(fullJsmd) {
   const jsmdLines = fullJsmd.split('\n')
   const chunks = []
   let currentChunkLines = []
   let currentEvalType = ''
   let evalFlags = []
   let currentChunkStartLine = 0
+  let chunkContent = ''
+
+  const newChunkId = (str) => {
+    const hash = hashCode(str)
+    let hashNum = '0'
+    for (const chunk of chunks) {
+      const [prevHash, prevHashNum] = chunk.chunkId.split('_')
+      if (hash === prevHash) {
+        hashNum = (parseInt(prevHashNum, 10) + 1).toString()
+      }
+    }
+    return `${hash}_${hashNum}`
+  }
 
   for (const [i, line] of jsmdLines.entries()) {
     if (line.slice(0, 2) === '%%') {
       // if there is any match, a new chunk has started
       // push the current chunk (unless it's on line 0), then reset
       if (i !== 0) {
         // DON'T push a chunk if we're only on line 0
+        chunkContent = currentChunkLines.join('\n')
```

Comment:
I think it would be better to declare the chunkContent as `const` here, instead of up around line 22.

Reply:
addressed with https://github.com/iodide-project/iodide/pull/1179/commits/48c94233566bead592f4ef1b0c5ac3cbd03b8b80